.travis.yml

```diff
@@ -51,6 +51,16 @@ env:
 
 jobs:
   include:
+    - name: Coding Standards
+      # stage: cs
+      php:
+        - 8.3
```

Comment:
What does this php version represent?

Reply:
The fixers run with this version. They run in a separate job without affecting the other jobs, e.g. Unittest.

Reply:
I think the CI script should splitted into stages.

- cs
- test
- docs
- build

In this way, the "docs" have their own setup and do not have to read an env value. Each stage can be handled separately ... and can fail separately

Reply:
Config file triggers a warning on Travis https://app.travis-ci.com/github/mantisbt/mantisbt/builds/269882010/config

```
[warn] on jobs.include.php: unexpected sequence, using the first value (8.3)
```

I think it should be just
```
      php: 8.3
```
But I'm wondering if it is even needed to specify the PHP version... Can't we simply rely on the worker's system default (I think it's 7.4 on focal, which we currently use)

Reply:
This is my first time with travis, but the config and the keys looks very similar to the gitlab ci config. So what I have done here is based on my experience with Gitlab.

> [warn] on jobs.include.php: unexpected sequence, using the first value (8.3)

Can not find this message on the linked page. I have tested the config with the travis config tool, no warings were shown.

Anyway `php: 8.3` looks good also … like the gitlab notation.


> But I'm wondering if it is even needed to specify the PHP version... Can't we simply rely on the worker's system default (I think it's 7.4 on focal, which we currently use) ? This way we don't have to worry about changing that in the future if and when PHP 8.3 becomes obsolete.

Not sure, will test it with the config tool. We have defined some PHP versions as global or shared versions for all jobs. Adding a key "php" to the job overrides the global/shared "php". Similar to neutralize the services `services: []` oder the `before_script: []` for this specific job. 

Simpler: Will the CS job run with all global defined PHP-Versions or not?
 
Please do not waste time on this, i will find the best working solution (without invisible :-) warnings)

Reply:
> > [warn] on jobs.include.php: unexpected sequence, using the first value (8.3)
> 
> Can not find this message on the linked page. I have tested the config with the travis config tool, no warings were shown.

Weird. Maybe an access rights issue caused by you not being member of mantisbt org ? I just checked and I it's still there
![image](https://github.com/mantisbt/mantisbt/assets/449891/8fe15b84-f9a5-4aef-94f3-6686c238fb00)

Reply:
Anyway it's gone from more recent builds, so the point is moot.
https://app.travis-ci.com/github/mantisbt/mantisbt/jobs/620323284/config